doc/metrics.md

```diff
@@ -13,65 +13,119 @@ relatedlinks: https://grafana.com/grafana/dashboards/15726
 
-In order for Prometheus to scrape from LXD, it has to be added to the targets.
+To do so, you can set either the [`core.metrics_address`](server-options-core) server configuration option or the [`core.https_address`](server-options-core) server configuration option.
+The `core.metrics_address` option is intended for metrics only, while the `core.https_address` option exposes the full API.
+So if you want to use a different address for the metrics API than for the full API, or if you want to expose only the metrics endpoint but not the full API, you should set the `core.metrics_address` option.
 
-First, one needs to ensure that `core.https_address` is set so LXD can be reached over the network.
-This can be done by running:
+For example, to expose the full API on the `8443` port, enter the following command:
 
-```bash
-lxc config set core.https_address ":8443"
+    lxc config set core.https_address ":8443"
+
+To expose only the metrics API endpoint on the `8444` port, enter the following command:
+
+    lxc config set core.metrics_address ":8444"
+
+To expose only the metrics API endpoint on a specific IP address and port, enter a command similar to the following:
+
+    lxc config set core.metrics_address "192.0.2.101:8444"
+
+### Add a metrics certificate to LXD
+
+Authentication for the `/1.0/metrics` API endpoint is done through a metrics certificate.
+A metrics certificate (type `metrics`) is different from a client certificate (type `client`) in that it is meant for metrics only and doesn't work for interaction with instances or any other LXD entities.
+
+To create a certificate, enter the following command:
+
+    openssl req -x509 -newkey ec -pkeyopt ec_paramgen_curve:secp384r1 -sha384 -keyout metrics.key -nodes -out metrics.crt -days 3650 -subj "/CN=metrics.local"
+
+```{note}
+The command requires OpenSSL version 1.1.0 or later.
 ```
 
-Alternatively, one can use `core.metrics_address` which is intended for metrics only.
+Then add this certificate to the list of trusted clients, specifying the type as `metrics`:
+
+    lxc config trust add metrics.crt --type=metrics
+
+### Make the metrics certificate available for Prometheus
 
-Second, the newly created certificate and key, as well as the LXD server certificate need to be accessible to Prometheus.
-For this, these three files can be copied to `/etc/prometheus/tls`:
+If you run Prometheus on a different machine than your LXD server, you must copy the required certificates to the Prometheus machine:
+
+- The metrics certificate (`metrics.crt`) and key (`metrics.key`) that you created
+- The LXD server certificate (`server.crt`) located in `/var/snap/lxd/common/lxd/` (if you are using the snap) or `/var/lib/lxd/` (otherwise)
+
+Copy these files into a `tls` directory that is accessible to Prometheus, for example, `/var/snap/prometheus/current/tls` (if you are using the snap) or `/etc/prometheus/tls`.
```

Comment:
I find the prometheus snap to be poorly maintained, they don't follow upstream very closely thus shipping outdated and known vulnerable versions :/. So I think we are better off keeping the non-snap version in our doc.

That said, providing the path for where to save the TLS stuff for snap users totally makes sense! I would however suggest to avoid the `current` directory as that one is replaced whenever the snap is updated (albeit not often!). As such, I'd go with `/var/snap/prometheus/common/tls` instead. That's what I was using when I tried the snap a while ago.

Reply:
Yeah, text looks fine, but agreed on using `common/tls` instead of`current/tls` here.

Reply:
The video uses the snap, so I think we need both.
I changed the TLS directory to `common/tls` now - but the `prometheus.yml` file doesn't exist in the common directory, so I left that file path as it is.